broker/src/main/java/io/zeebe/broker/system/partitions/ZeebePartition.java

```diff
@@ -169,20 +169,14 @@ private void onRoleChange(final Role newRole, final long newTerm) {
   }
 
   private CompletableFuture<Void> onRaftFailed() {
-    final CompletableFuture<Void> inactiveTransitionFuture = new CompletableFuture<>();
+    final CompletableFuture<Void> future = new CompletableFuture<>();
     actor.run(
         () -> {
-          final ActorFuture<Void> transitionComplete = transitionToInactive();
-          transitionComplete.onComplete(
-              (v, t) -> {
-                if (t != null) {
-                  inactiveTransitionFuture.completeExceptionally(t);
-                  return;
-                }
-                inactiveTransitionFuture.complete(null);
-              });
+          handleUnrecoverableFailure();
```

Comment:
Raft failures are not always unrecoverable. It could be due to any unhandled exception, which might go away after a restart. It is true that it is not automatically recovered. But a simple restart might fix it. Do we want to mark such errors as unrecoverable, or only errors which are truly unrecoverable such as corrupted state?